internal/api/parsers/update.go

```diff
@@ -0,0 +1,117 @@
+package parsers
+
+import (
+	"awans.org/aft/internal/api/operations"
+	"awans.org/aft/internal/db"
+	"fmt"
+)
+
+func (p Parser) ParseUpdate(oldRec db.Record, args map[string]interface{}) (op operations.UpdateOperation, err error) {
+	unusedKeys := make(set)
+	for k := range args {
+		unusedKeys[k] = void{}
+	}
+
+	data := p.consumeData(unusedKeys, args)
+	newRec, nested, err := p.update(oldRec, data)
+	if err != nil {
+		return
+	}
+
+	include, err := p.consumeInclude(oldRec.Interface().Name(), unusedKeys, args)
+	if err != nil {
+		return
+	}
+
+	if len(unusedKeys) != 0 {
+		return op, fmt.Errorf("%w: %v", ErrUnusedKeys, unusedKeys)
+	}
+	op = operations.UpdateOperation{Old: oldRec, New: newRec, Nested: nested, Include: include}
+	return op, err
+}
+
+func updateRecordFromData(oldRec db.Record, keys set, data map[string]interface{}) (db.Record, set, error) {
```

Comment:
looking at this a little more -- I'm not sure it really makes sense to be doing this work in the "parser" -- this is really applying the operation! We've already fetched the data and constructed the new record before we're even done "parsing." Seems out of order to me. 

Can we push this part into the operation?